blog/articles/distributed-tracing.md

```diff
@@ -0,0 +1,488 @@
+collector.
+
+Finally we can configure Eventing and Serving to point all of their traces to
+our collector:
+
+
+```
+for ns in knative-eventing knative-serving; do
+  kubectl patch --namespace "$ns" configmap/config-tracing \
+   --type merge \
+   --patch '{"data":{"backend":"zipkin","zipkin-endpoint":"http://otel-collector.observability:9411/api/v2/spans", "debug": "true"}}'
+done
+```
+
+
+The `debug` flag here tells Knative to send _all_ traces to our collector,
+whereas in a real-world deployment you’d probably want to set a sample rate to
+only get a representative subset of your traces.
+
+
+## Hello, world?
+
+Now that our tracing infrastructure is all deployed and configured, we can start
+taking advantage of it by deploying some services. We can deploy [the heartbeat
+image](https://github.com/knative/eventing/blob/41e36eea88aada0f73e98ff5fb39b6509135dda1/cmd/heartbeats/main.go)
+as a
+[ContainerSource](https://knative.dev/docs/eventing/samples/container-source/)
+to test and see that everything is wired up correctly:
+
+
+```
+kubectl apply -f - <<EOF
+apiVersion: sources.knative.dev/v1
+kind: ContainerSource
+metadata:
+  name: heartbeats
+spec:
+  template:
+    spec:
+      containers:
+        - image: benmoss/heartbeats:latest
+          name: heartbeats
+          args:
+            - --period=1
+          env:
+            - name: POD_NAME
+              value: "heartbeats"
+            - name: POD_NAMESPACE
+              value: "default"
+            - name: K_CONFIG_TRACING
+              value: '{"backend":"zipkin","debug":"true","sample-rate":"1","zipkin-endpoint":"http://otel-collector.observability:9411/api/v2/spans"}'
+  sink:
+    uri: http://dev.null
+EOF
+```
+
+For right now this container will just be sending its heartbeats to the
+non-existent domain [http://dev.null](http://dev.null), so if we look at the
+logs for this pod we will see a bunch of DNS resolution errors. However if we
+inspect the logs of our `otel-collector` pod, we should see that it’s
+successfully receiving traces from our service. It’s good confirmation that our
+configuration works, but not very exciting from a tracing perspective! Let’s
+make it somewhat realistic by adding in a Knative service to receive our
+heartbeats:
+
+
+```
+kubectl apply -f - <<EOF
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: event-display
+spec:
+  template:
+    spec:
+      containers:
+        - image: benmoss/event_display:latest
+          env:
+            - name: K_CONFIG_TRACING
+              value: '{"backend":"zipkin","debug":"true","zipkin-endpoint":"http://otel-collector.observability:9411/api/v2/spans"}'
+EOF
+```
+
+We'll update our heartbeats service to start sending heartbeats here instead:
+
+```
+kubectl apply -f - <<EOF
+apiVersion: sources.knative.dev/v1
+kind: ContainerSource
+metadata:
+  name: heartbeats
+spec:
+  template:
+    spec:
+      containers:
+        - image: benmoss/heartbeats:latest
```

Comment:
Can we use the upstream image instead of the personal image?
```suggestion
        - image: gcr.io/knative-releases/knative.dev/eventing/cmd/heartbeats
```